### PR TITLE
New lua function for style `BGRotationOffset`

### PR DESCRIPTION
**Pull request by Zly-u**
Added Lua function for the background rotation uh... Because why not lol.
Didn't seem like anyone did mind it, so.

### CODE REVIEW

File: src/SSVOpenHexagon/Core/HGScripting.cpp [hunk 1]
```diff
@@ -451,6 +451,12 @@ void HexagonGame::initLua_StyleControl()
     sdVar("3dPerspectiveMult", &StyleData::_3dPerspectiveMult);
     sdVar("BGTileRadius", &StyleData::bgTileRadius);
     sdVar("BGColorOffset", &StyleData::BGColorOffset);
+    sdVar("BGRotationOffset", &StyleData::BGRotOff);
+
+    addLuaFn("s_setColorOffset",
+        [this](int mValue) {styleData.BGColorOffset = mValue;})
+        .arg("value")
+        .doc("Offsets the colors of the background tiles by `$0`.");
```
Comment (SuperV1234) [inline src/SSVOpenHexagon/Core/HGScripting.cpp:459]:
```suggestion
```

File: include/SSVOpenHexagon/Data/StyleData.hpp [hunk 2]
```diff
@@ -48,6 +48,7 @@ class StyleData
     float _3dPerspectiveMult;
     float bgTileRadius{4500.f};
     unsigned int BGColorOffset{0};
+    float BGRotOff{0};
```
Comment (SuperV1234) [inline include/SSVOpenHexagon/Data/StyleData.hpp:51]: Degrees? Radians? Pixels? A comment would be nice :)

Comment (SuperV1234) [inline include/SSVOpenHexagon/Data/StyleData.hpp:51]:
```suggestion
    float BGRotOff{0}; // In degrees
```

Comment (Zly-u) [reply include/SSVOpenHexagon/Data/StyleData.hpp]: Degrees in lua, they get converted in the code to Radians to fit the calculations.

**Pull request merged**